cstar/base/additional_code.py

```diff
@@ -47,8 +32,8 @@ def __init__(
         self,
         location: str,
         subdir: str = "",
-        checkout_target: str | None = None,
-        files: list[str] | None = None,
+        checkout_target: str = "",
+        files: list[str] = [],
```

Comment:
defaulting to a mutable list is a bug and will result in accumulation errors when called multiple times